frontend/src/users/components/UserActivation.tsx

```diff
@@ -0,0 +1,98 @@
+import { useEffect, useState } from 'react';
+import { AxiosError } from 'axios';
+import { validate as uuidValidate } from 'uuid';
+import { Badge, Spinner, Text, VStack } from '@chakra-ui/react';
+import { ApplicationErrorDto } from '@shared/dtos/application-error';
+import { ApplicationErrorCode } from '@shared/enums/error-code';
+import { HttpStatus } from '@shared/enums/http';
+import { applicationErrorHandler } from '@shared/helpers/application-error';
+import { useActivate } from '@users/hooks/api';
+import { UserActivationParams } from '@users/types/activation';
+
+interface ActivationStatusDto {
+  validUuids: boolean;
+  loading: boolean;
+  error: boolean;
+  errorData?: ApplicationErrorDto<ApplicationErrorCode, HttpStatus>;
+}
+export const UserActivation: React.FC<UserActivationParams> = ({ userId, activationToken }) => {
+  const { mutate: activate } = useActivate();
+  const isUuidValid = uuidValidate(userId) && uuidValidate(activationToken);
+  const [activationStatus, setActivationStatus] = useState<ActivationStatusDto>({
+    validUuids: isUuidValid,
+    loading: isUuidValid,
+    error: false,
+  });
+
+  const handleSuccess = (): void => {
+    setActivationStatus((state) => ({
+      ...state,
+      loading: false,
+      error: false,
+    }));
+  };
+
+  const handleError = (error: AxiosError<ApplicationErrorDto<any, any>, unknown>): void =>
+    applicationErrorHandler<UserActivationParams>()
+      .onGenericValidationFailed(({ details }) => console.log(details))
```

Comment:
IMHO niepotrzebny ten console.log

Reply:
czy takie cos jest mozliwe do zrobienia?
`.onGenericValidationFailed(({ details }) => handleAppliactionError(mapValidationErrors(details)))`

mapValidationErrors zwraca
`type 'Record<string, string>'`

Reply:
Nie rozumiem. Jesli `handleAppliactionError` przyjmuje stringa, a `mapValidationErrors` zwraca rekord, to typy sie nie zgadzaja